### PR TITLE
User role code for blog six.

### PR DESCRIPTION
**Pull request by rick-kenna**

### CODE REVIEW

File: python/user_roles/update_roles.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    
+    # Obtain the mapping dictionaries
+    user_name_to_id = get_user_ids(base_url, headers)
+    role_name_to_id = get_role_ids(base_url, headers)
+
+    # Read each row in the CSV file and process. Skip over comments.
+    with open(csv_file_name, 'r') as reader_obj:
+        csv_reader = reader(reader_obj)
+        for row in csv_reader:
+            if row[0].startswith('#'):
```
Comment (sfgeorge) [inline python/user_roles/update_roles.py:123]: Nifty support for comments! 👍

File: python/user_roles/update_roles.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        print("API key is non-existent")
+        sys.exit(1)
+    
+    # HTTP headers.
+    headers = {'X-Risk-Token': api_key,
+               'Content-Type': 'application/json; charset=utf-8',
+               'User-Agent': 'sample.connector_ui/1.0.0 (Cisco Secure)'}
+    
+    # You might have to change this depending on your deployment.
+    base_url = "https://api.kennasecurity.com/"
```
Comment (sfgeorge) [inline python/user_roles/update_roles.py:113]: optional thing – not necessary – but we could make this the default base_url, but overrideable via an optional argument or environment variable.

**Review by sfgeorge — APPROVED** [review]
Looks awesome.  Added a nice-to-have that isn't needed, just while struggling to come up with ideas to make it better.  But this looks great and doesn't need further changes, at least from my own view.  nice job!

File: python/user_roles/update_roles.py (hunk 2 above)
Comment (rick-kenna) [reply python/user_roles/update_roles.py:113]: This is a good idea.  I will add it to the next blog.

**Pull request merged**